File: src/main/groovy/com/moowork/gradle/node/yarn/YarnExecRunner.java
package com.moowork.gradle.node.yarn;

import com.moowork.gradle.node.exec.ExecRunner;
import org.gradle.api.Project;
import org.gradle.process.ExecResult;

import java.io.File;


public class YarnExecRunner extends ExecRunner {

	public YarnExecRunner(final Project project) {
		super(project);
	}

	@Override
	protected ExecResult doExecute() {
		return run(this.variant.getYarnExec(), this.getArguments());
	}

	@Override
	protected String computeAdditionalBinPath() {
		if (this.ext.getDownload()) {
			String yarnBinDir = this.variant.getYarnBinDir().getAbsolutePath();
			String npmBinDir = this.variant.getNpmBinDir().getAbsolutePath();
			String nodeBinDir = this.variant.getNodeBinDir().getAbsolutePath();
			return yarnBinDir + File.pathSeparator + npmBinDir + File.pathSeparator + nodeBinDir;
		}

		return null;
	}
}

File: src/main/kotlin/com/github/gradle/node/yarn/YarnExecRunner.kt
package com.github.gradle.node.yarn

import com.github.gradle.node.exec.ExecRunner
import org.gradle.api.Project
import org.gradle.process.ExecResult
import java.io.File

class YarnExecRunner(project: Project) : ExecRunner(project) {

    override fun doExecute(): ExecResult {
        return run(variant.yarnExec, arguments)
    }

    override fun computeAdditionalBinPath(): String {
        if (!ext.download) {
            return ""
        }
        val yarnBinDir = variant.yarnBinDir.absolutePath
        val npmBinDir = variant.npmBinDir.absolutePath
        val nodeBinDir = variant.nodeBinDir.absolutePath
        return yarnBinDir + File.pathSeparator + npmBinDir + File.pathSeparator + nodeBinDir
    }
}
